package-res/resources/web/pentaho/data/filter/AbstractTreeFilter.js

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./AbstractFilter",
+  "../../util/object",
+  "../../lang/ArgumentRequiredError",
+  "./_apply",
+  "./_toSpec"
+], function(AbstractFilter, O, ArgumentRequiredError, _apply, toSpec) {
+  "use strict";
+
+  /**
+   * @name AbstractTreeFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @extends pentaho.data.filter.AbstractFilter
+   * @abstract
+   * @amd pentaho/data/filter/AbstractTreeFilter
+   *
+   * @classdesc The `AbstractTreeFilter` class is the abstract base class of
```

Comment:
@classdesc The abstract base class for filters that aggregate the outcome of other filters.